src/main/java/hudson/remoting/Launcher.java

```diff
@@ -539,6 +543,11 @@ private SSLSocketFactory getSSLSocketFactory()
                     throw x;
                 } else
                     throw e;
+            } catch (FileNotFoundException e) {
+                System.err.println("Failing to obtain "+slaveJnlpURL);
+                e.printStackTrace(System.err);
+                System.err.println("Will silently exit without errors");
+                System.exit(0);
```

Comment:
I am not sure that Exiting without errors && with the zero error code is fine here To be reviewed

Reply:
The idea after this zero exit code is to take advantage of KostyaSha/yet-another-docker-plugin#186: Defining a docker template with a restart policy (on-failure) would keep the agent alive while it is registered at the Jenknins master.
Once (If ever) it gets de-registered, this modified code would stop retrying and let the container die (wont be deleted)

Nevertheless, if anyone finds an objection, the exit code could be setup via a new flag. I would keep in mind that this PR tries to cover a case where agents are being erased after a Jenkins restart, while the agent itself is/will be back alive (leaving zombie agents trying to reconnect). As the real issue is almost unreproducible, this PR is about a defensive behaviour.